### PR TITLE
Deprecate generic.checksums.luhn and ean

### PR DESCRIPTION
**Pull request by benkonrath**
I've been meaning to do this for a while. As the code suggests, my plan is to remove the `generic.checksums.luhn` and `generic.checksums.ean` functions in 3.0 - users should just use python-stdnum directly.

@claudep If you have time a review and your thoughts on this plan would be helpful. Thanks.

### CODE REVIEW

File: tests/test_deprecated.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+from localflavor.generic.checksums import luhn
+
+try:
+    from unittest.mock import patch
+except ImportError:
+    from mock import patch
+
+
+class DeprecatedFieldsTests(SimpleTestCase):
+    @patch('localflavor.generic.checksums.stdnum_luhn')
```
Comment (claudep) [inline tests/test_deprecated.py]: Not sure, what is the role of patch here?

Comment (claudep) [reply tests/test_deprecated.py]: Oh, for the assertion below? All that just for testing a deprecation. I wouldn't have moaned if you omitted that :-)

File: tests/test_deprecated.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    def test_luhn_deprecated(self, stdnum_luhn):
+        with warnings.catch_warnings(record=True) as recorded:
+            warnings.simplefilter('always')
+            luhn('1234')
+
+        self.assertTrue(all(w.category is RemovedInLocalflavor30Warning for w in recorded))
+        stdnum_luhn.asert_called_once_with('1234')
```
Comment (claudep) [inline tests/test_deprecated.py]: Strange that the tests don't fail with the typ0 `asert` vs `assert`

Comment (benkonrath) [reply tests/test_deprecated.py]: That's mock - it silently allows undefined methods to be called without raising an error.

File: tests/test_deprecated.py (hunk 1 above)
Comment (benkonrath) [reply tests/test_deprecated.py]: I agree it's a bit pedantic - thanks for pointing this out. Removing these tests also means I can remove the additional requirement for mock on Python 2 which simplifies things a bit.

File: tests/test_deprecated.py (hunk 2 above)
Comment (benkonrath) [reply tests/test_deprecated.py]: By the way - nice catch! :-)

**Pull request merged**